### DISCUSSION TITLE
Why the peculiar choice of session cookies?

### DISCUSSION THREAD

**Discussion by zehawki** [medusajs/medusa#2203] [Q&A] [answered]
I'm unable to understand why a choice has been made to "stay logged in" via session cookies. With REST API all I need is a sessionless token - opaque or JWT. The whole point of headless is that this is one part of a larger system, and therefore I might have another system dealing with end user (aka customer for medusa) auth. And therefore the customer will not log _into_ the medusa system, but elsewhere and I just need an endpoint to create that customer in medusa, without using email/pwd creds, get back a token, And then exercise whichever medusa API I need via that customer token. Session cookies also make load balancing a headache.

In the docs, I see this: [How to Add API Token to a User](https://docs.medusajs.com/api/admin/#section/Authentication/api_token), which seems to be only for users (ie admin) and not for customers. And anyway there is no endpoint.

At this point, I see no way to proceed to integrate Medusa into a larger system because there seems to be a basic assumption that the medusa backend works in isolation.

Am I correct in my understanding so far? And if so, is there any plan on changing this?

**Comment by srindom** [comment, marked as answer]
@zehawki - this is an excellent point! We will revise the whole auth flow and make it more agnostic to how you authenticate and which methods work for API authorization. We don't have it planned for the next 3 months but would start looking into it beginning of next year.

**Reply by zehawki** [reply]
Next year.... 😯

**Reply by sergioruizsan** [reply]
just passing by through this interesting question and wanted to appreciate all the GREAT work and effort you are doing @srindom ! Will wait for it or try to contribute :)

**Reply by zehawki** [reply]
@srindom any progress on this?

**Comment by viewer2018** [comment]
您好!   您的邮件我已收到,我会尽快查看!kind regards,何尾风

**Comment by nkmnz** [comment]
Hi everyone!
Are there plans to put this feature on the roadmap? :)

**Comment by viewer2018** [comment]
您好!   您的邮件我已收到,我会尽快查看!kind regards,何尾风
